src/Servers/Kestrel/shared/KnownHeaders.cs

```diff
@@ -131,14 +131,18 @@ static KnownHeaders()
                 "Connection",
                 "Server",
                 "Date",
-                "Transfer-Encoding"
+                "Transfer-Encoding",
+                "Keep-Alive",
+                "Upgrade"
             };
             var enhancedHeaders = new[]
             {
                 "Connection",
                 "Server",
                 "Date",
-                "Transfer-Encoding"
+                "Transfer-Encoding",
+                "Keep-Alive",
+                "Upgrade"
```

Comment:
Or here, they already exist in other lists. Why not make a Clear method for all known headers?